README.md

```diff
@@ -196,6 +198,43 @@ const view = (state, actions) =>
 
 It is important to understand that the result of the view is a new Virtual DOM. It may seem wasteful to throw away the old Virtual DOM and re-create it entirely on every update — not to mention the fact that at any one time, Hyperapp is keeping two Virtual DOM trees in memory, but as it turns out, browsers can create hundreds of thousands of objects very quickly. On the other hand, modifying the DOM is several orders of magnitude more expensive.
 
+## Mounting
+
+> **Work In Progress**: This section is not perfect and may be incomplete. Please help.
+
+Hyperapp will mount your rendered view into the given <samp>container</samp> Element. The default behavior is non-destructive. We will not take over any existing elements inside the container.
+
+```js
+app(state, actions, view, container)
+```
+
+### SSR
```

Comment:
I'd like to think more about this section, but I don't think we need to hold up this PR on account of that. 
 
The thing that's confusing to me, is that even if hyperapp *didn't* do hypdration proper (the `toVnode` thing), but *only* the replaced the 5th-argument-element with itself, we'd still have the "time-to-interactive" benefit described here.

So for one, the `replaceElement`-argument is a feature in itself. You could have simply `<div id="myapp">App not started yet</div>` as the replace-element.

For two, what benefit does the fact that hyperapp calls `toVnode` behind the scenes provide (over just replacing the dom that was already there)? Something with lifecycle events? Anyway, whatever it is, we should clarify it.

And depending on what the difference is, maybe we want to split the `### SSR`  section into two.

Reply:
Oh never mind. I get it. `toVnode`  just makes the first render less expensive *if* the existing DOM matches the app's first VDOM well enough.  Right? 

In that case I don't think we need to mention the word "hydration" or "SSR" at all (to me those are scary words that I associate with complex and framework-specific server-side setup). "Pre-existing elements" or even "pre-existing html" sound more approachable, and are afaik equally correct.

Reply:
Hydration doesn't fix the "uncanny valley" situation where before it has completed the app is visible but not functional.  It does at least leave whatever the user may have input and not cause possible reflow/repaint/flicker/scroll resets etc.  Others will have to defend why they want it as is since I haven't been actively using hyperapp myself atm.

Reply:
@andyrj To clarify: I wasn't saying we should remove the hydration feature. Rather that we should not *call it* "hydration", because it makes it seem more magical than it really is.

It took me until today to realize that what we're really talking about is simply replacing static html that is already there, with whatever the app renders.  The fact that we use `toVnode` is an internal optimization of that (less dom-operations to do the replacing) that the user doesn't need to know about. From the outside it simply looks like "replacing existing html".

Reply:
Hydration is about adopting current nodes it isn't "replacing" anything other than event handlers.

Reply:
Maybe call it adopt()?

Reply:
Also I think that time to interactive was from me when @JorgeBucaran was asking me to help him clarify the point of it back some months ago now.  If someone has a less buzzwordy description I definitely think that bit could be improved upon...  Have I told you all I am terrible at documentation yet? 😉

Reply:
@andyrj If your preexisting dom is `<div id="app-mount-point">App not started, yet...</div>` then when your app starts, it will replace that with the app UI. But there will be no reusing of elements since the existing DOM does not match the app's vdom closely enough.

From a users perspective, it's replacing what was there, regardless of wether hyperapp is able to repurpose those elements or not.

Reply:
> @andyrj Have I told you all I am terrible at documentation yet? 😉

Here we go again. Just kidding. 🤣